.github/workflows/ci.yml

```diff
@@ -48,7 +48,20 @@ jobs:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
       - uses: astral-sh/setup-uv@v5
       - run: uv tool install poethepoet
-      - run: uv sync --all-extras
+      - run: uv sync --all-extras --python ${{ matrix.python }}
+      - name: Check python version
+        shell: bash
+        run: |
+          actual=$(uv run python -c "import sys; print('.'.join(map(str, sys.version_info[:2])))")
+          if [[ "$actual" != "${{ matrix.python }}" ]]; then
+            echo "Python version in use by uv ($actual) does not match intended version (${{ matrix.python }})"
+            exit 1
+          fi
+          actual=$(python -c "import sys; print('.'.join(map(str, sys.version_info[:2])))")
+          if [[ "$actual" != "${{ matrix.python }}" ]]; then
+            echo "Python version on PATH ($actual) does not match intended version (${{ matrix.python }})"
+            exit 1
+          fi
```

Comment:
Is this embedded GH workflow logic/check necessary? Can we not trust `setup-python`? Do we need to check Rust version, protoc version, CPU arch, etc too if we're concerned with the actions? I can't tell from the PR description whether something happened that made this not work and made us concerned.